public/types/index.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+export interface DocumentsIndex {
+  'docs.count': string;
+  'docs.deleted': string;
+  health: 'green' | 'yellow';
+  index: string;
+  pri: string;
+  'pri.store.size': string;
+  rep: string;
+  status: 'open' | 'close';
+  'store.size': string;
+  uuid: string;
+}
+
+export interface IDocType {
+  [key: string]: string;
+}
```

Comment:
When we test this, we'll need to make sure to include test cases for complex field types like [nested](https://opensearch.org/docs/latest/opensearch/supported-field-types/nested/), as well as [arrays](https://opensearch.org/docs/latest/opensearch/supported-field-types/index/#arrays) of simple values.

Not a blocker for the initial commit, IMO, but definitely something we'll need to include in testing.